src/main/java/teammates/logic/api/GateKeeper.java

```diff
@@ -327,21 +327,29 @@ public void verifyAccessibleForCurrentUserAsInstructorOrTeamMember(AccountAttrib
             String section, String email) {
         InstructorAttributes instructor = instructorsLogic.getInstructorForGoogleId(courseId, account.googleId);
         if (instructor != null) {
-            if (!instructor.isAllowedForPrivilege(section,
-                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
-                throw new UnauthorizedAccessException("Instructor does not have enough privileges to view the photo");
-            }
+            verifyInstructorCanViewPhoto(instructor, section);
             return;
         }
-
         StudentAttributes student = studentsLogic.getStudentForCourseIdAndGoogleId(courseId, account.googleId);
         if (student != null) {
-            if (!studentsLogic.isStudentsInSameTeam(courseId, email, student.email)) {
-                throw new UnauthorizedAccessException("Student does not have enough privileges to view the photo");
-            }
+            verifyStudentCanViewPhoto(student, email, courseId);
             return;
         }
+        throw new UnauthorizedAccessException("User does not have enough privileges to view the photo");
+    }
+
+    private boolean verifyInstructorCanViewPhoto(InstructorAttributes instructor, String section) {
```

Comment:
This is supposed to be a void that does throwing, otherwise calling it has no effect.